### PR TITLE
AdminJS Dev workspaces

### PR DESCRIPTION
**Pull request by ariansobczak-rst**
Introduced monorepo for AdminJS dev

Related changes in submodules to run monorepo (PRs)
[adminjs](https://github.com/SoftwareBrothers/adminjs/pull/1017)
[adminjs-hapijs](https://github.com/SoftwareBrothers/adminjs-hapijs/pull/43)
[adminjs-design-system](https://github.com/SoftwareBrothers/adminjs-design-system/pull/23)
[adminjs-upload](https://github.com/SoftwareBrothers/adminjs-upload/pull/38)
[adminjs-sequelizejs](https://github.com/SoftwareBrothers/adminjs-sequelizejs/pull/96)
[adminjs-typeorm](https://github.com/SoftwareBrothers/adminjs-typeorm/pull/36)
[adminjs-example-app](https://github.com/SoftwareBrothers/adminjs-example-app/pull/33)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**/node_modules/**
```
Comment (dziraf) [inline .eslintignore:1]: Shouldn't we also add submodule paths here?

File: .gitmodules [hunk 2]
```diff
@@ -1,72 +1,77 @@
...
 	url = git@github.com:SoftwareBrothers/adminjs-fastify.git
 	branch = main
+[submodule "better-docs"]
+	path = better-docs
+	url = git@github.com:SoftwareBrothers/better-docs.git
+	branch = master
+[submodule "packages/adminjs-bundler"]
+	path = packages/adminjs-bundler
+	url = git@github.com:SoftwareBrothers/adminjs-bundler.git
+    branch = main
```
Comment (dziraf) [inline .gitmodules:76]: too small indent?

File: infrastructure/Dockerfile [hunk 3]
```diff
@@ -2,12 +2,12 @@ FROM node:10.13.0
...
 WORKDIR /usr/src/app
 
-COPY ["adminjs/package*.json", "adminjs/yarn.lock", "./adminjs/"]
-COPY ["adminjs-hapijs/package*.json", "adminjs-hapijs/yarn.lock" , "./adminjs-hapijs/"]
-COPY ["adminjs-expressjs/package*.json", "adminjs-expressjs/yarn.lock", "./adminjs-expressjs/"]
-COPY ["adminjs-mongoose/package*.json", "adminjs-mongoose/yarn.lock" , "./adminjs-mongoose/"]
-COPY ["adminjs-sequelizejs/package*.json", "adminjs-sequelizejs/yarn.lock", "./adminjs-sequelizejs/"]
-COPY ["adminjs-example-app/package*.json", "adminjs-example-app/yarn.lock", "./adminjs-example-app/"]
+COPY ["packages/adminjs/package*.json", "packages/adminjs/yarn.lock", "./adminjs/"]
+COPY ["packages/dminjs-hapijs/package*.json", "packages/adminjs-hapijs/yarn.lock" , "./adminjs-hapijs/"]
```
Comment (dziraf) [inline infrastructure/Dockerfile:6]: typo -> `packages/dminjs-hapijs`

File: scripts/bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+# Build all packages
+echo -e "${YELLOW}Building packages...${NO_COLOR}"
+# Build mongoose first in order to be dependencey of @adminjs/import-export
+yarn workspace @adminjs/mongoose build
+yarn workspaces --no-progress run build
+
+# Build AdminJS
+echo -e "${YELLOW}Building AdminJS...${NO_COLOR}"
+yarn workspace adminjs bundle:globals
```
Comment (dziraf) [inline scripts/bootstrap.sh:29]: I'm not sure if `build` and `bundle:globals` generates only production bundle or both dev and prod?

File: .gitmodules (hunk 2 above)
Comment (ariansobczak-rst) [reply .gitmodules:76]: Fixed by increasing indentation

File: infrastructure/Dockerfile (hunk 3 above)
Comment (ariansobczak-rst) [reply infrastructure/Dockerfile:6]: Fixed

File: scripts/bootstrap.sh (hunk 4 above)
Comment (ariansobczak-rst) [reply scripts/bootstrap.sh:29]: I am not sure if I understood well :) Here `build` and `bundle:globals` is to initialize (bootstrap) and is run without `NODE_ENV=production` so generate `global-bundle.development.js` bundle (if you eg want to just run an example app). If you would work on admin js in dev mode then you should run `bundle` (is then in dev mode with reloads)

**Review by dziraf — APPROVED** [review]

File: .eslintignore (hunk 1 above)
Comment (ariansobczak-rst) [reply .eslintignore:1]: Added todo and exclude packages from linting

**Pull request merged**